common/v2/services/Store/Account/AccountProvider.tsx

```diff
@@ -86,14 +87,55 @@ export const AccountProvider: React.FC = ({ children }) => {
         state.updateAccount(existingAccount.uuid, existingAccount);
       }
     },
-    updateAccountsBalances: toUpdate => {
+    updateAllAccountsAssets: async (storeAccounts, assets) =>
```

Comment:
Can you try refactoring using lenses? Basically this is what we want to do:
```
const people = [{
  name: 'Terry',
  cats: ['Korin', 'Sweep', 'Caterina']
}, {
  name: 'Trevor',
  cats: ['Jazz']
}]
const cats = R.lensProp('cats')
const uppercaseCats = R.map(R.over(cats, R.map(R.toUpper)))
uppercaseCats(people)
> [{
    name: 'Terry',
    cats: ['KORIN', 'SWEEP', 'CATERINA']
  }, {
    name: 'Trevor',
    cats: ['JAZZ']
  }]
```
https://itnext.io/a-beginners-guide-to-ramda-part-2-lenses-62bdd3993598

Reply:
I've been scratching my head over how to do this and realize that there may be something wrong with implementing it this way. Currently `updateAllAccountsAssets` doesn't take in `storeAccounts` and update `StoreProvider`'s `storeAccounts`, which is what I think this change would be helpful for. It currently takes in `storeAccounts` and updates `AccountProvider`'s `accounts` which `StoreProvider` is watching for changes to (in a `useMemo`), then updates `storeAccounts`. 

I didn't change that functionality at all, i just changed how many times we update `AccountProvider`'s `accounts` by swapping from `model.update` to `model.updateAll`, and swapping from just scanning all tokens to scanning tokens for 1 account in relevant places.

I think this change would be a good one to make, but it'd be a larger change and should probably be it's own task.